Scripts/RemediationScripts/Remediate-EnableAADForKubernetesService.ps1

```diff
@@ -0,0 +1,586 @@
+                        @{Expression={$_.EnableRBAC};Label="Is RBAC enabled on Kubernetes Cluster?";Width=20;Alignment="left"},
+                        @{Expression={$_.IsAADEnabled};Label="Is AAD enabled on Kubernetes Cluster?";Width=20;Alignment="left"}
+
+        Write-Host "`nFollowing Kubernetes cluster(s) are having AAD disabled:" -ForegroundColor $([Constants]::MessageType.Info)
+        
+        $kubernetesServicesWithoutAADEnabled | Format-Table -Property $colsProperty -Wrap
+  
+        Write-Host $([Constants]::SingleDashLine)
+        Write-Host "AAD will be enabled for all Kubernetes Services."
+        Write-Host "`nRBAC must be enabled on Kubernetes cluster." -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host "Once AAD is enabled, you won't be able to disable again." -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host "For more information, please refer: https://docs.microsoft.com/en-us/azure/aks/azure-ad-rbac" -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host "`nDo you want to enable AAD for all Kubernetes Services? " -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+            
+        $userInput = Read-Host -Prompt "(Y|N)"
+
+        if($userInput -ne "Y")
+        {
+            Write-Host "AAD will not be enabled for any Kubernetes Services. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+            break
+        }
+
+        Write-Host "`nDo you want to add Azure AD groups as administrators on each cluster?" -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+        
+        $addAADGroup = $false
+        $input = Read-Host -Prompt "(Y|N)"
+
+        if($input -eq "Y")
+        {
+            $addAADGroup = $true
+        }
+        
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "`n[Step 5 of 5] Enabling AAD for Kubernetes Services..."
+
+        # To hold results from the remediation.
+        $kubernetesClusterRemediated = @()
+        $kubernetesClusterSkipped = @()
+
+        $setAADProfile = [SetAADProfile]::new()
+
+        $kubernetesServicesWithoutAADEnabled | ForEach-Object {
+            $kubernetesServiceResource = $_
+            $resourceId = $_.Id
+            $resourceGroupName = $_.ResourceGroupName
+            $resourceName = $_.Name
+            $location = $_.Location
+            $isRBACEnabled = $_.EnableRBAC
+            $aadClientIds = @()
+
+            Write-Host $([Constants]::SingleDashLine)
+            Write-Host "`nEnabling AAD for Kubernetes cluster [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+
+            # Check whether RBAC is enabled on cluster or not. Without RBAC, AAD cannot be enabled.
+            if ($isRBACEnabled -eq $true)
+            {
+                if ($addAADGroup)
+                {
+                    Write-Host "Please provide Azure AD Group object id: "
```

Comment:
Can we provide multiple group object Id?

Reply:
Also Just add that this group will be added as cluster admin,
 Write-Host "Please provide Azure AD group object id (group will be registered as an admin group on the cluster): "

Reply:
Multiple groups can be added but we have supported one group (optional) using BRS.

Reply:
ok,